### PR TITLE
YG-16: Update missing filed into the form, Change DatetimePicker to DatePiker. Optimize code callback function

### PR DESCRIPTION
**Pull request by phienngo**
YG-16: Update missing filed into the form, Change DatetimePicker to DatePiker. Optimize code callback function

### CODE REVIEW

File: medkumo.js [hunk 1]
```diff
@@ -1,3 +1,274 @@
 /*
-* Contain logic function
-*/
+ * Medkumo Javascript SDK
+ *
+ */
+(function(window, undefined) {
+    var Config = {},
+        params = {};
```
Comment (tailehuu) [inline medkumo.js]: params = getUrlParameters();

File: sdk.js [hunk 2]
```diff
@@ -12,8 +12,9 @@
 
     Medkumo.book = function(hospitalKey, doctorKey) {
         console.log('executing book...');
-        var session= new Date();
-        PopupCenter('index.html?hospitalKey=' + hospitalKey + '&doctorKey=' + doctorKey + '&session=' + session, "Book An Appointment", 650, 500);
+        var date = new Date();
+        var session = date.getDate() + "" + date.getMonth() + "" + date.getFullYear();
+        PopupCenter('index.html?hospitalKey=' + hospitalKey + '&doctorKey=' + doctorKey + '&session=' + session, "Book An Appointment", 650, 600);
```
Comment (tailehuu) [inline sdk.js]: could we rename it to openPopup(...)

File: medkumo.js (hunk 1 above)
Comment (phienngo) [reply medkumo.js]: ok, I will fix it.

File: sdk.js (hunk 2 above)
Comment (phienngo) [reply sdk.js]: yes

File: medkumo.js [hunk 3]
```diff
@@ -1,3 +1,283 @@
...
+
+        // events
+        bookAnAppointmentEvents();
+    }
+
+    function bookAnAppointmentEvents() {
+        $(".medkumo_datepicker").datepicker({
+            dateFormat: 'd-m-yy'
+        });
+        var currentDate = new Date().toISOString();
```
Comment (tailehuu) [inline medkumo.js]: move this var to top

File: medkumo.js [hunk 4]
```diff
@@ -1,3 +1,283 @@
...
+            }
+            //14/12/2016
+            if (appointmentDateValue != null) {
+                appointmentDate = appointmentDateValue.getDate() + "/" + (appointmentDateValue.getMonth() + 1) + "/" + appointmentDateValue.getFullYear();
+            }
+            if (dobValue != null) {
+                dob = dobValue.getDate() + "/" + (dobValue.getMonth() + 1) + "/" + dobValue.getFullYear();
+            }
+
+            var jsonData = {
```
Comment (tailehuu) [inline medkumo.js]: all declare variable should be on top

File: medkumo.js [hunk 5]
```diff
@@ -1,3 +1,283 @@
...
+        }
+    };
+    // validation functions
+    function isValidEmailAddress(emailAddress) {
+        var pattern = /^([a-z\d!#$%&'*+\-\/=?^_`{|}~\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]+(\.[a-z\d!#$%&'*+\-\/=?^_`{|}~\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]+)*|"((([ \t]*\r\n)?[ \t]+)?([\x01-\x08\x0b\x0c\x0e-\x1f\x7f\x21\x23-\x5b\x5d-\x7e\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]|\\[\x01-\x09\x0b\x0c\x0d-\x7f\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]))*(([ \t]*\r\n)?[ \t]+)?")@(([a-z\d\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]|[a-z\d\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF][a-z\d\-._~\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]*[a-z\d\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])\.)+([a-z\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]|[a-z\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF][a-z\d\-._~\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]*[a-z\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])\.?$/i;
+        return pattern.test(emailAddress);
+    };
+
+    function validateBookAnAppointment() {
+        var result = true;
```
Comment (tailehuu) [inline medkumo.js]: var result = true,
       patientName = ....

File: medkumo.js (hunk 3 above)
Comment (phienngo) [reply medkumo.js]: yes sir.

File: medkumo.js (hunk 4 above)
Comment (phienngo) [reply medkumo.js]: yes

File: medkumo.js (hunk 5 above)
Comment (phienngo) [reply medkumo.js]: yes.

File: sdk.js [hunk 6]
```diff
@@ -12,11 +12,12 @@
 
     Medkumo.book = function(hospitalKey, doctorKey) {
         console.log('executing book...');
-        var session= new Date();
-        PopupCenter('index.html?hospitalKey=' + hospitalKey + '&doctorKey=' + doctorKey + '&session=' + session, "Book An Appointment", 650, 500);
+        var date = new Date();
```
Comment (tailehuu) [inline sdk.js]: var date = new Date(),
session

**Comment by tailehuu** [comment]
pls rename this index.html to booking.html. Thanks

**Comment by tailehuu** [comment]
+1 let's merge

**Pull request merged**